### ISSUE TITLE
A Simple Framework for Contrastive Learning of Visual Representations

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#24]
## [A Simple Framework for Contrastive Learning of Visual Representations](https://arxiv.org/abs/2002.05709)
*This paper presents SimCLR: a simple framework for contrastive learning of visual representations. We simplify recently proposed contrastive self-supervised learning algorithms without requiring specialized architectures or a memory bank. In order to understand what enables the contrastive prediction tasks to learn useful representations, we systematically study the major components of our framework. We show that (1) composition of data augmentations plays a critical role in defining effective predictive tasks, (2) introducing a learnable nonlinear transformation between the representation and the contrastive loss substantially improves the quality of the learned representations, and (3) contrastive learning benefits from larger batch sizes and more training steps compared to supervised learning. By combining these findings, we are able to considerably outperform previous methods for self-supervised and semi-supervised learning on ImageNet. A linear classifier trained on self-supervised representations learned by SimCLR achieves 76.5% top-1 accuracy, which is a 7% relative improvement over previous state-of-the-art, matching the performance of a supervised ResNet-50. When fine-tuned on only 1% of the labels, we achieve 85.8% top-5 accuracy, outperforming AlexNet with 100X fewer labels.*

### Cite this paper
```bibtex
@misc{chen2020simple,
    title={A Simple Framework for Contrastive Learning of Visual Representations},
    author={Ting Chen and Simon Kornblith and Mohammad Norouzi and Geoffrey Hinton},
    year={2020},
    eprint={2002.05709},
    archivePrefix={arXiv},
    primaryClass={cs.LG}
}
```

**Comment by juanmc2005** [comment]
### Description

This paper introduces a simple contrastive self-supervised learning method mostly based on image augmentation techniques.They use ResNet as encoder, followed by a non-linear projection head, whose outputs are used to calculate a modified cross entropy loss.

Unsupervised training is done on the ImageNet ILSVRC-2012 dataset, and evaluation is performed on downstream tasks by fine-tuning a linear classification layer with frozen encoder parameters.

### Loss

The loss is calculated over all positive pairs in a batch, generated by augmenting a single image in two different ways. Negative pairs consist of an example of the positive pair together with all other examples in the batch.

They use the cosine similarity of pairs as softmax's input divided by a temperature parameter to control de "sharpness" of the output probability distribution. They call this loss NT-Xent (for normalized temperature-scaled cross entropy loss).

### Results

There are a couple interesting results that might be useful in other domains (not only computer vision):

1. Composition of multiple data augmentations leads to better performance than individual transformations.
2. Contrastive self-supervised learning needs more data augmentation than the supervised counterparts.
3. Contrastive self-supervised learning benefits more from bigger models.
4. The non-linear projection head improves the quality of representations in the layer before it.
5. Contrastive self-supervised learning benefits more from larger batch sizes and training.

It is surprising that applying the loss after the non-linear head improves the quality of the penultimate layer's representations. They explain this by the fact that the loss is invariant to augmentation, so the last activations remove this information. However, that can actually be helpful for other downstream task that can rely on color, position, etc.

### Takeaway

I would say that the fact that self-supervised models benefit more from bigger models, batch sizes and training is to be expected, as the lack of a task-specific training signal complicates the encoding function to approximate. This ultimately leads to needing more data (hence the benefit of augmentation) and more complex models (hence bigger models and longer training). On the other hand, as they clearly explain, big batch sizes improve performance because more negative examples exist per batch, improving the training signal.

In general, I think the biggest takeaway of this paper is the idea of using augmentation as the main driver for a self-supervised training signal, wich can also be applied to speech, say for a specific audio chunk. Their loss function is not that novel, as CPC makes use of a very similar loss.

I would say this is kind of an augmentation driven CPC, which is definitely a good idea, and might be helpful for us wherever augmentation can be applied (easy in speech, more difficult in NLP).
